lib/wibox/layout/fixed.lua

```diff
@@ -42,8 +47,16 @@ function fixed:layout(context, width, height)
             pos = pos + w + spacing
         end
 
-        if (self._private.dir == "y" and pos-spacing > height) or
-            (self._private.dir ~= "y" and pos-spacing > width) then
+        -- Add the spacing widget
+        if k > 1 and abspace > 0 and spacing_widget then
+            table.insert(result, base.place_widget_at(
+                spacing_widget, is_x and (x - spoffset) or x, is_y and (y - spoffset) or y,
+                is_x and abspace or w, is_y and abspace or h
+            ))
+        end
+
+        if (is_y and pos-spacing > height) or
+            (is_x and pos-spacing > width) then
             break
```

Comment:
Why is this check after "add a spacing widget" and not before it? If the layout contains so many widgets that not all of them can be displayed, there will be a spacing widget after the last visible widget and that spacing widget will be outside of the space assigned to this `fixed` layout.
Also, you might want to think about what should happen in this situation. I'd say a spacing widget should only be shown if there is `spacing` space for it (or if `spacing` is negative...). If e.g. only half of `spacing` is available, then no widget should be shown. But that's just an idea, feel free to come up with other ones. (Dunno if we should have tests for the behaviour in this case.)